internal/engine/live_update_build_and_deployer.go

```diff
@@ -159,20 +159,22 @@ func (lubad *LiveUpdateBuildAndDeployer) buildAndDeploy(ctx context.Context, cu
 	}()
 
 	if len(toArchive) > 0 {
-		l.Infof("Will copy %d file(s) to container: %s", len(toArchive), cInfo.ContainerID.ShortStr())
+		l.Infof("Will copy %d file(s) to container(s): %s", len(toArchive), cIDStr)
 		for _, pm := range toArchive {
 			l.Infof("- %s", pm.PrettyStr())
 		}
 	}
 
-	err = cu.UpdateContainer(ctx, cInfo, pr, build.PathMappingsToContainerPaths(toRemove), boiledSteps, hotReload)
-	if err != nil {
-		if build.IsUserBuildFailure(err) {
-			return WrapDontFallBackError(err)
+	for _, cInfo := range state.RunningContainers {
+		err = cu.UpdateContainer(ctx, cInfo, pr, build.PathMappingsToContainerPaths(toRemove), boiledSteps, hotReload)
+		if err != nil {
```

Comment:
i think we need to keep updating all the containers so they're in a consistent state?

i feel like when there are multiple containers, we probably need a much more aggressive fallback strategy than when there's one container

Reply:
if one update fails, they definitionally won't be in a consistent state -- and the effect here would be that ALL the targets for this manifest fall back to the Image Build and Deployer. (idk what happens if we succeed at one container update and fail at the next, fall back to ibad, and THAT fails... but this is all i can think of right now? any other ideas?)

Reply:
you said "ALL the targets for this manifest fall back"

did you see the line "WrapDontFallBackError" two lines below this? i'm 90% sure that doesn't fall back

Reply:
okay, mulled this over some, have some thoughts. (first of all, we should rename that function -- it means that a user's _run_ step failed. but whatever.) lmk what you think:

1. a container update fails, it's NOT a `UserRunFailure` -- e.g. we couldn't connect to the synclet or something. We fall back to image build. Doesn't matter if any container updates succeeded previously.
2. a container update fails with a `UserRunFailure` (e.g. `go build` failed b/c syntax error), run the container update on all remaining containers -- if there's a syntax error and all of the updates fail, we at least want the same fs state on all the containers
2a. if all the updates fail with `UserRunFailure`, great, don't fall back. (We can assume that it's the same failure -- which is probably good enough for now)
2b. if any update _succeeds_ (or exits with `!UserRunFailure`), we have potentially disparate state, fall back to image build.